packages/back-end/src/scim/users/putUser.ts

```diff
@@ -0,0 +1,98 @@
+import { cloneDeep } from "lodash";
+import { Response } from "express";
+import { ScimError, ScimUser, ScimUserPutRequest } from "../../../types/scim";
+import { expandOrgMembers } from "../../services/organizations";
+import { updateOrganization } from "../../models/OrganizationModel";
+import { MemberRole, OrganizationInterface } from "../../../types/organization";
+import { isRoleValid } from "./createUser";
+
+async function updateUserRole(
+  org: OrganizationInterface,
+  userId: string,
+  newRole: MemberRole
+) {
+  const updatedOrgMembers = cloneDeep(org.members);
+
+  const userIndex = org.members.findIndex((member) => member.id === userId);
```

Comment:
Add sanity check here to make sure the user exists in the members array

Reply:
10-4!